srcpkgs/Signal-Desktop/template

```diff
@@ -1,26 +1,40 @@
 # Template file for 'Signal-Desktop'
 pkgname=Signal-Desktop
-version=1.39.5
+version=5.0.0
 revision=1
 # Due to electron
 # 32-bit is not supported https://github.com/signalapp/Signal-Desktop/issues/1661
 archs="x86_64"
-hostmakedepends="git nodejs python tar yarn"
-depends="libnotify libappindicator"
+hostmakedepends="git git-lfs nodejs python3 tar yarn openssl"
+depends="gtk+3 libnotify libappindicator libvips"
 short_desc="Signal Private Messenger for Linux"
 maintainer="Julio Galvan <juliogalvan@protonmail.com>"
-license="GPL-3.0-only"
+license="AGPL-3.0-only"
 homepage="https://github.com/signalapp/Signal-Desktop"
 distfiles="https://github.com/signalapp/Signal-Desktop/archive/v${version}.tar.gz"
-checksum=f9f55c0195579b43712bf370b753337432eda4bcf89809d1c8f3deb0b4ee4400
+checksum=4ff58f5ab5f4de6477dd92069d8c2e9a830513acca527b51e0ee1c7cbe7d8627
 nostrip_files="signal-desktop"
 
 pre_build() {
-	vsed -e 's/"node": "/&>=/' -i package.json
+	# git-lfs hook needs to be installed for sqlcipher dependency 
+	git-lfs install
+
+	# `yarn cache clean` needed for sqlcipher dependency to install.
+	# Unfortunately, because we build from tarball we can't do this
+	# step at the beginning but only after we generate node_modules
+	# See: https://github.com/signalapp/Signal-Desktop/issues/5032
+	yarn install --ignore-engines || rm -r node_modules; yarn cache clean
+	yarn install --frozen-lockfile
+
+	# Have SQLCipher dynamically link from OpenSSL
+	# See https://github.com/signalapp/Signal-Desktop/issues/2634
+	patch --forward --strip=1 --input="${FILESDIR}/openssl-linking.patch"
+
+	# We can't read the release date from git so we use SOURCE_DATE_EPOCH instead
+	patch --forward --strip=1 --input="${FILESDIR}/expire-from-source-date-epoch.patch"
```

Comment:
The patches should go in `patches` and left to be applied automatically. This is an arch artifact.

Reply:
believe me, I thought so too! but actually the files to be patched don't yet exist just after the extract, they need to be patched at this point

Reply:
I was able to leave out the epoch source date patch